ui/app/hooks/useTransactionDisplayData.js

```diff
@@ -0,0 +1,124 @@
+ * @typedef {Object} TransactionDisplayData
+ * @property {string} title               - primary description of the transaction
+ * @property {string} subtitle            - supporting text describing the transaction
+ * @property {string} category            - the transaction category
+ * @property {string} primaryCurrency     - the currency string to display in the primary position
+ * @property {string} [secondaryCurrency] - the currency string to display in the secondary position
+ * @property {string} status              - the status of the transaction
+ * @property {string} senderAddress       - the Ethereum address of the sender
+ * @property {string} recipientAddress    - the Ethereum address of the recipient
+ */
+
+/**
+ * useTransactionDisplayData
+ *
+ * The goal of this method is to perform all of the necessary computation and
+ * state access required to take a transactionGroup and derive from it a shape
+ * of data that can power all views related to a transaction. Presently the main
+ * case is for shared logic between transaction-list-item and transaction-detail-view
+ * @param {Object} transactionGroup group of transactions
+ * @return {TransactionDisplayData}
+ */
+export function useTransactionDisplayData (transactionGroup) {
+  const knownTokens = useSelector(tokenSelector)
+  const t = useI18nContext()
+  const { initialTransaction, primaryTransaction } = transactionGroup
+  // initialTransaction contains the data we need to derive the primary purpose of this transaction group
+  const { transactionCategory } = initialTransaction
+
+  const { from: senderAddress, to } = initialTransaction.txParams || {}
+
+  // for smart contract interactions, methodData can be used to derive the name of the action being taken
+  const methodData = useSelector((state) => getKnownMethodData(state, initialTransaction?.txParams?.data)) || {}
+
+  const actionKey = getTransactionActionKey(initialTransaction)
+  const statusKey = getStatusKey(primaryTransaction)
+  const status = statusKey in pendingStatusMap ? 'pending' : statusKey
+
+  const primaryValue = primaryTransaction.txParams?.value
+  let prefix = '-'
+  const date = formatDateWithYearContext(initialTransaction.time || 0)
+  let subtitle = date
+  let recipientAddress = to
+
+  const token = knownTokens.find((token) => token.address === recipientAddress)
+  const tokenData = useTokenData(initialTransaction?.txParams?.data)
+  const tokenDisplayValue = useTokenDisplayValue(initialTransaction?.txParams?.data, token)
+
+  let category
+  let title
+  // There are four types of transaction entries that are currently differentiated in the design
+  // 1. Send (sendEth sendTokens)
+  // 2. Deposit
+  // 3. Site interaction
+  // 4. Approval
+  if (transactionCategory == null) {
+    category = SIGNATURE_REQUEST
+    title = t('signatureRequest')
+    subtitle += ` · ${initialTransaction.msgParams?.origin || initialTransaction.origin || ''}`
+  } else if (transactionCategory === 'approve') {
```

Comment:
All of these action categories are declared as constants in `ui/app/helpers/constants/transactions.js`. Better to use the constants here instead of the string literals.